tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bats
+# Copyright (c) 2023 Intel Corporation
+# Copyright (c) 2023 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+    [[ "${PULL_TYPE}" =~ "guest-pull" ]] ||  skip "Test only working for pulling image inside the guest"
+    setup_common
```

Comment:
Hey Chengyu, with respect to the test, I think we need some of the logic from the other confidential test: https://github.com/kata-containers/kata-containers/blob/61ce7455c51689b72138a00c4b8c0c6130c2500f/tests/integration/kubernetes/k8s-confidential.bats#L13-L25 in setup and https://github.com/kata-containers/kata-containers/blob/61ce7455c51689b72138a00c4b8c0c6130c2500f/tests/integration/kubernetes/k8s-confidential.bats#L57-L59 in teardown to ensure that the tests are only run for supported hypervisors (the confidential ones) and if testing with non-tee that we add the image annotation to switch to use the confidential image.

I guess the best approach would be to refactor this into lib.sh as confidential_setup/confidential_teardown or similar?

Reply:
Done.